cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/VerificationStage.java

```diff
@@ -72,7 +77,14 @@ public void process(ApplicationDeployable input) throws Exception {
     ApplicationSpecification specification = input.getSpecification();
     Id.Application appId = input.getId();
 
-    // TODO: add a check against system applications (adapters, for instance).
+    if (ApplicationDeployScope.USER.equals(input.getApplicationDeployScope())) {
+      AdapterTypeInfo adapterTypeInfo = adapterService.getAdapterTypeInfo(appId.getId());
+      if (adapterTypeInfo != null) {
+        throw new RuntimeException
```

Comment:
Ok, I just left it as RuntimeException - because the other application verifier throws RuntimeException upon verification failure (just 4 lines below).